src/Microsoft.Unity.Analyzers/ProtectedUnityMessage.cs

```diff
@@ -0,0 +1,111 @@
+/*--------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE in the project root for license information.
+ *-------------------------------------------------------------------------------------------*/
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.Unity.Analyzers.Resources;
+
+namespace Microsoft.Unity.Analyzers
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class ProtectedUnityMessageAnalyzer : DiagnosticAnalyzer
+	{
+		internal static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+			id: "UNT0021",
+			title: Strings.ProtectedUnityMessageDiagnosticTitle,
+			messageFormat: Strings.ProtectedUnityMessageDiagnosticMessageFormat,
+			category: DiagnosticCategory.Correctness,
+			defaultSeverity: DiagnosticSeverity.Info,
+			isEnabledByDefault: true,
```

Comment:
After discussing with @jbevain, we decided to mark this one with "isEnabledByDefault: false". It means that this analyzer will be `opt-in`. To enable it, users will need to add the following to their `.editorconfig` file:

```editorconfig
[*.cs]
dotnet_diagnostic.UNT0021.severity = suggestion
```

We will also update VS/VSMac tooling to offer `Protected` scope for generated messages.